lsr_role2collection.py

```diff
@@ -688,6 +688,21 @@ def file_replace(path, find, replace, file_patterns):
                 with open(filepath, "w", encoding="utf-8") as f:
                     f.write(s)
 
+def main_file_exists(src_path, role):
+    filenames = ["main.yml", "main.yaml"]
+    paths = [
+        os.path.join(src_path, "tasks"),
+        os.path.join(src_path, role, "tasks")
+    ]
+
+    for path in paths:
+        for filename in filenames:
+            logging.debug(f"Testing if  {os.path.join(path, filename)} is a file.")
+            if os.path.isfile(os.path.join(path, filename)):
+                logging.debug(f"{os.path.join(path, filename)} exists and is a file - setting src_path to {path}.")
+                src_path = os.path.join(path)
```

Comment:
Setting `src_path` here does not set `src_path` in the calling context.  Maybe have this function return `src_path` if the main YAML file exists, or raise an error if the file does not exist?